### PR TITLE
[JNI] Add setKernelPinnedCopyThreshold and setPinnedAllocationThreshold

### PR DESCRIPTION
**Pull request by abellina**
In 24.08 two new cuDF methods are being added, and the second method is still in flight (see: https://github.com/rapidsai/cudf/pull/16206):

```
cudf::set_kernel_pinned_copy_threshold
cudf::set_allocate_host_as_pinned_threshold
```

We'd like to expose these methods in our JNI layer. I created a Cudf.java with the two static methods, and put the definitions in CudfJni.cpp. 

Marked as draft since I need https://github.com/rapidsai/cudf/pull/16206 to merge, and we are still testing it.

### CODE REVIEW

**Comment by abellina** [comment]
Update: this PR is still not going to build, but I would like to make sure we consider it for 23.08 if https://github.com/rapidsai/cudf/pull/16206 goes in.

**Comment by vyasr** [comment]
Realistically I think this PR is going to have to slip at this point. We might get #16206 in but I don't think we'll have time to iterate through getting this updated and ready in time.

**Comment by abellina** [comment]
> Realistically I think this PR is going to have to slip at this point. We might get #16206 in but I don't think we'll have time to iterate through getting this updated and ready in time.

This PR is pretty small and it's only in the JNI side of things, I am hoping we can get it in, but I don't think we absolutely need it for 24.08.

**Review by revans2 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/Cudf.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+public class Cudf {
+
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/Cudf.java:20]:
```suggestion

```

File: java/src/main/java/ai/rapids/cudf/Cudf.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+public class Cudf {
+    
+  static {
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/Cudf.java:21]:
```suggestion
public class Cudf {

  static {
```

**Comment by abellina** [comment]
I have retriggered the python tests as they seem to be failing with connection issues (unrelated). I am OK if this PR slips to 24.10.

**Pull request merged**